frb/figures/galaxies.py

```diff
@@ -198,26 +200,45 @@ def sub_sfms(ax_M, galaxies, clrs, markers):
         else:
             continue
         # SFR
-        if 'SFR_nebular_err' in galaxy.derived.keys():
-            logS, sig_logS = utils.log_me(galaxy.derived['SFR_nebular'], galaxy.derived['SFR_nebular_err'])
+        # Try Nebular first
+        if 'SFR_nebular' in galaxy.derived.keys():
+            if 'SFR_nebular_err' in galaxy.derived.keys():
+                logS, sig_logS = utils.log_me(galaxy.derived['SFR_nebular'], galaxy.derived['SFR_nebular_err'])
+            else:
+                logS, sig_logS = utils.log_me(galaxy.derived['SFR_nebular'], 0.3*galaxy.derived['SFR_nebular'])
+        elif 'SFR_photom' in galaxy.derived.keys():
+            # Demand 3 sigma
+            logS, sig_logS = utils.log_me(galaxy.derived['SFR_photom'], galaxy.derived['SFR_photom_err'])
+            if galaxy.derived['SFR_photom'] < 3*galaxy.derived['SFR_photom_err']:
+                sig_logS = None
         else:
-            logS, sig_logS = utils.log_me(galaxy.derived['SFR_nebular'], 0.3*galaxy.derived['SFR_nebular'])
+            continue
+        # Color
+        if isinstance(clrs, str):
+            assert clrs == 'auto'
+            if sig_logS is None:
+                clr = 'red'
+            else:
+                clr = 'blue'
+        else:
+            clr = clrs[kk]
         # Plot
         ax_M.errorbar([logM], [logS], xerr=sig_logM, yerr=sig_logS,
-                      color=clrs[kk], marker=markers[kk],
+                      color=clr, marker=markers[kk],
                      markersize="12", capsize=3, label=galaxy.name)
         if sig_logS is None:
             # Down arrow
-            plt.arrow(logM, logS, 0., -0.2, fc=clrs[kk], ec=clrs[kk],
+            plt.arrow(logM, logS, 0., -0.2, fc=clr, ec=clr,
                   head_width=0.02*4, head_length=0.05*2)
 
     ax_M.annotate(r"\textbf{Star forming}", (8.5, 0.8), fontsize=13.)
     ax_M.annotate(r"\textbf{Quiescent}", (11, -0.9), fontsize=13.)
     ax_M.set_xlabel("$\log \, (M_*/M_\odot)$")
     ax_M.set_ylabel("$\log \, SFR (M_\odot$/yr)")
-    ax_M.legend(loc='lower left')
+    if show_legend:
+        ax_M.legend(loc='lower left')
     ax_M.set_xlim(7.5, 11.8)
-    ax_M.set_ylim(-2.5, 1.2)
+    ax_M.set_ylim(-2.4, 1.6)
```

Comment:
- Just to be clear: Does this code take the SFR based on H-alpha first and then only use SFR_photom if SFR_nebular is none? I think it should be that way. 
- I also considered whether we should use the SDSS emission catalog instead? That contains stellar masses, plus we can infer SFR from the listed H-alpha flux + redshift (so more directly comparable to the FRB hosts).
- Also, can we change the plot to instead of giving each galaxy a separate color, color the galaxies as known repeaters / so-far non-repeaters?

Reply:
Yes, SFR_nebular first, then SFR_photom.

Having SDSS as an option would be good.  Especially for z~0 FRBs.

As for the repeating, this is not a parameter included in any of
our FRBs (yet).  But we can add that one of these days.